src/main/java/com/sergtm/service/impl/HeartRateServiceImpl.java

```diff
@@ -95,36 +96,50 @@ public void addHeartRateById(Long id, int upperPressure, int lowerPressure, int
     @Transactional
     public boolean deleteHeartRate(Long id) {
         HeartRate heartRate = heartRateDao.getById(id);
-        if(heartRate==null){
+        if (heartRate == null) {
             return false;
         }
         try {
             heartRateDao.deleteHeartRate(heartRate);
             return true;
-        }catch (HibernateException e){
+        } catch (HibernateException e) {
 
         }
         return false;
     }
 
-    private HeartRate createAndSaveHeartRate(int upperPressure, int lowerPressure, int beatsPerMinute,Date datetime, Person person) {
-        HeartRate hr = HeartRate.createHeartRate(upperPressure, lowerPressure, beatsPerMinute,datetime, person);
+    private HeartRate createAndSaveHeartRate(int upperPressure, int lowerPressure, int beatsPerMinute, Date datetime, Person person) {
+        HeartRate hr = HeartRate.createHeartRate(upperPressure, lowerPressure, beatsPerMinute, datetime, person);
         heartRateDao.addHeartRate(hr);
         return hr;
     }
 
     @Override
-    public Collection<StatisticOnDay> getChartData(Long personId) {
+    public Collection<StatisticOnDay> getChartData(Long personId, String from, String to) {
         LocalDate now = LocalDate.now();
+        LocalDateTime fromDate;
+        LocalDateTime toDate;
         LocalDateTime firstDayOfMonth = now.withDayOfMonth(1).atStartOfDay();
         LocalDateTime lastDayOfMonth = now.withDayOfMonth(now.lengthOfMonth()).atTime(23, 59);
 
+        if (from == null || from.equals("")) {
+            fromDate = firstDayOfMonth;
+        } else {
+            fromDate = LocalDateTime.parse(from);
+        }
+
+        if (to == null || to.equals("")) {
```

Comment:
Create a function parseDate with default value. Something like that:
public  Date parseDate(Date parsed, Date defaultValue)

Move it to static DateUtil class

Reply:
Done